pkg/cloud/cloud_test.go

```diff
@@ -203,6 +203,30 @@ func TestCreateDisk(t *testing.T) {
 			expErr:               fmt.Errorf("could not create volume in EC2: CreateVolume generic error"),
 			expCreateVolumeErr:   fmt.Errorf("CreateVolume generic error"),
 		},
+		{
+			name:       "fail: ec2.CreateVolume returned snapshot not found error",
+			volumeName: "vol-test-name-error",
+			diskOptions: &DiskOptions{
+				CapacityBytes:    util.GiBToBytes(1),
+				Tags:             map[string]string{VolumeNameTagKey: "vol-test", AwsEbsDriverTagKey: "true"},
+				AvailabilityZone: expZone,
+			},
```

Comment:
i'm confused by this test case. don't you need a snapshot ID in the diskOptions to get a Snapshot not found error?

Reply:
It's testing [cloud.go](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/pkg/cloud/cloud.go#L350) so we don't need to actually provide a valid snapshotId, this test case just expect go sdk to return snapshot not found fault code to the client and see if client can handle the error properly.

Reply:
i see. we're just checking that the error gets handled. makes sense.